Adding molecule tests

neilhaggerty opened the pull request:
Also moving dependencies into the task as this is the "new" syntax.

Keeping the geerlingguy.pip dependency in meta/ as we want to pin the
version.

PLAT-1595

Commit:
WIP

Commit:
Adding Molecule tests.

Also moving dependencies into the task as this is the "new" syntax.

Keeping the geerlingguy.pip dependency in meta/ as we want to pin the
version.

PLAT-1595

tomyan commented on molecule/default/molecule.yml:
```diff
@@ -0,0 +1,30 @@
+platforms:
+  - name: centos_image
+    image: centos:7
+    privileged: true
```
Is privileged necessary for these two contianers?

neilhaggerty replied:
Have removed

grahamlyons commented on meta/main.yml:
```diff
@@ -27,27 +28,26 @@ galaxy_info:
-
-dependencies:
-    - mergermarket.pip
-    - mergermarket.docker
```
We run the agent in Docker so we need Docker to be installed so we need our Ansible role for Docker, no?

grahamlyons replied:
Just re-read your description on the pull request about the "new" syntax for dependencies. Does that install them just-in-time then?

neilhaggerty replied:
Going to move these back and let users of the role decide on how to manage dependencies

tomyan commented on molecule/default/tests/test_default.py:
```diff
@@ -0,0 +1,32 @@
+
+
+def test_datadog_docker_image_exists(host):
+    output = host.check_output('docker images datadog/agent -q | wc -l')
```
This check seems a little inexact - would match 1 or 947372723282184834, but not 947372723282784834.

neilhaggerty replied:
I don't understand this

tomyan replied:
Was just saying (in a bit of a round about way) that it should check it's one not that it contains the digit 1 somewhere.

neilhaggerty replied:
ohhhh... I didn't realise I was using `in`.. I meant `==` good catch. Have fixed now

... was a little cryptic though haha

grahamlyons commented on molecule/default/requirements.yml:
```diff
@@ -0,0 +1,6 @@
+---
+# from galaxy
+- src: geerlingguy.repo-epel
```
We've not got this in the dependencies for the role. Do you know what we pull from EPEL?

grahamlyons reviewed (changes requested):
Looks great. Just got a couple of questions.

tomyan commented on test.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+docker run --rm -it \
+    -v $(pwd):/${PWD##*/}:ro \
```
I find `${PWD##*/}` a little cryptic. Maybe split out into a temporary variable with a descriptive name?

tomyan commented on test.sh:
```diff
@@ -0,0 +1,8 @@
+docker run --rm -it \
+    -v $(pwd):/${PWD##*/}:ro \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -w /${PWD##*/} \
```
...would also DRY this out a little.

neilhaggerty replied:
don't understand this

tomyan reviewed (approved):
Minor comments inline

grahamlyons commented on molecule/default/tests/test_default.py:
```diff
@@ -0,0 +1,32 @@
+
+def test_datadog_docker_image_exists(host):
+    output = host.check_output('docker images datadog/agent -q | wc -l')
+    assert '1' in output
```
Perhaps `output.strip() == '1'`?

grahamlyons replied:
Or:
```python
output = host.check_output('docker images datadog/agent -q')
lines = output.split('\n')
assert len(lines) == 1
```

tomyan replied:
Or:

    lines = int(host.check_output('docker images datadog/agent -q | wc -l'))
    assert lines == 1

Commit:
Refactoring

Moving the dependencies to be in the `meta/` folder so they aren't
automatically installed as part of a role, and the user of the role has
to decide if they want to add the dependencies themselves.
